api/src/test/java/org/openmrs/module/datafilter/impl/UserProgramBasedFilterTest.java

```diff
@@ -116,4 +128,18 @@ public void getUsers_shouldReturnAllUsersIfTheFilterIsDisabled() {
 		assertEquals(7, getUsers().size());
 	}
 	
+	@Test(expected = None.class)
+	public void getProvidersByPerson_shouldNotFailWithSQLSyntaxErrorException() throws Exception {
+		// Setup
+		String sql = "DELETE FROM datafilter_entity_basis_map";
+		FilterTestUtils.enableFilter(ImplConstants.PROGRAM_BASED_FILTER_NAME_USER);
+		DatabaseUtil.executeSQL(getConnection(), sql, false);
```

Comment:
I thought you should run the query after the call to logout because you are testing that the query does not fail if there is no authenticated user